readthedocs/api/v3/views.py

```diff
@@ -50,7 +55,7 @@ class APIv3Settings:
     # Using only ``TokenAuthentication`` for now, so we can give access to
     # specific carefully selected users only
     authentication_classes = (TokenAuthentication,)
-    permission_classes = (PublicDetailPrivateListing,)
+    permission_classes = (IsAuthenticated & (ListCreateProject | PublicDetailPrivateListing),)
```

Comment:
This is starting to make me nervous and I don't really understand how `ListCreateProject` is that different from `PublicDetailPrivateListing`.

Reply:
Ouch! I thought that splitting these permissions in different classes would be easier to understand them.

Before this PR, everything was under a black box called `PublicDetailPrivateListing`. I decided to split it because it did more than what it name says.

My idea is:

- `IsAuthenticated` is the first **required** condition,

then, the user has access to

- `ListCreateProject`: `list` OR `create` at the `/projects/` endpoint (which is kind special)

> NOTE: `ListCreateProject` may be moved to `ProjectsViewSet` since it's the only place where it's needed

OR

- PublicDetailPrivateListing: `detail` on any object or `list` of "private" (personal) objects.

Let me know if this helps to clarify or if you have another idea how to make it better.

Reply:
I think overall, I'd feel a lot better if these special cases were not on the base class that every API endpoint extends from. The project endpoint is special (although with all the extra complexity, I'm starting to doubt that it should be) and these changes should be localized to there!